tasks/permissions.py

```diff
@@ -0,0 +1,36 @@
+from rest_framework.permissions import BasePermission
+from django.http import Http404
+
+from user.models import CustomUser
+
+SAFE_METHODS = ['GET', 'PATCH', 'HEAD', 'OPTIONS']
+
+class IsAdmin(BasePermission):
+   
+    def has_permission(self, request, view):
+        try:
+            role = request.user.role
+        except Exception:
+            raise Http404()
+        is_true = (role == CustomUser.ADMIN)
+        return bool(is_true and request.user.is_authenticated)
```

Comment:
This condition is unnecessary, If the user isn't authenticated then it will be in the exception block and never reach upto this point